src/app/donations/dontations.spec.js

```diff
@@ -0,0 +1,3 @@
+describe('prx.donations', function () {
+  beforeEach(module('prx.donations'));
```

Comment:
To test the service you extract should be pretty straightforward.

At that point, the responsibilities of the directive should be pretty straightforward to test. Mock out the service to test the various conditions. Do you know how to compile the directive without the DOM?